scipy/sparse/linalg/tests/test_propack.py

```diff
@@ -108,130 +108,32 @@ def test_svdp(ctor, precision, irl, which):
-
-    line_indptr = n_header
-    line_indices = line_indptr + n_indptr
-    line_data = line_indices + n_indices
-
-    def _concatenate_lines(lines):
-        return "".join([line.rstrip() for line in lines])
-
-    indptr = _concatenate_lines(contents[line_indptr:line_indices])
-    indptr = np.asarray([int(i) for i in indptr.split()])-1
-
-    indices = _concatenate_lines(contents[line_indices:line_data])
-    indices = np.asarray([int(i) for i in indices.split()])-1
-
-    data = _concatenate_lines(contents[line_data:])
-    data = np.asarray([float(data[i:i+datum_length])
-                       for i in range(0, len(data), datum_length)])
-    real, imag = data[::2], data[1::2]
-    data = real + imag*1.0j
-    data.astype(dtype)
-
-    return csc_matrix((data, indices, indptr), (m, n))
-
-
-def load_sigma(folder, precision="double", irl=False):
-    dtype = _dtype_map[precision]
-    s_name = "Sigma_IRL.ascii" if irl else "Sigma.ascii"
-    path = os.path.join(folder, precision, 'Examples', 'Output', s_name)
-    pydtype = {
-        np.complex64: complex,
-        np.complex128: complex,
-        np.float32: float,
-        np.float64: float,
-    }[dtype]
-    with open(path) as f:
-        data = np.array([pydtype(line.split()[1]) for line in f]).astype(dtype)
-    return data
-
-
-def load_uv(folder, precision="double", uv="U", irl=False):
-    dtype = _dtype_map[precision]
-    uv_name = (uv + "_IRL.ascii") if irl else (uv + ".ascii")
-    path = os.path.join(folder, precision, 'Examples', 'Output', uv_name)
-    with open(path) as f:
-        m, n = (int(val) for val in f.readline().split())
-        if precision in {'single', 'double'}:
-            data = np.array([dtype(val.strip()) for val in f], dtype=dtype)
-        else:
-            data = np.loadtxt(
-                path, dtype=dtype, skiprows=1, delimiter='\n',
-                converters={0: lambda s: complex(
-                    *[float(n) for n in s.decode()[1:-1].split(',')])})
-    return data.reshape((n, m)).T
-
-
 @pytest.mark.parametrize('precision', _dtype_testing)
 @pytest.mark.parametrize('irl', (False, True))
 def test_examples(precision, irl):
     atol = {
-        'single': 1e-3,
+        'single': 1e-4,
         'double': 1e-9,
         'complex8': 1e-4,
         'complex16': 1e-9,
     }[precision]
 
     path_prefix = os.path.dirname(__file__)
-    relative_path = ['..', '_propack', 'PROPACK']
-    folder = os.path.join(path_prefix, *relative_path)
-
-    A = {
-        'single': load_real,
-        'double': load_real,
-        'complex8': load_complex,
-        'complex16': load_complex,
-    }[precision](folder, precision)
-    s_expected = load_sigma(folder, precision, irl=irl)
-    u_expected = load_uv(folder, precision, "U", irl=irl)
-    vh_expected = load_uv(folder, precision, "V", irl=irl).T
-
-    k = len(s_expected)
-    u, s, vh, _ = _svdp(A, k, irl_mode=irl, random_state=0)
+    relative_path = "propack_test_data.npz"
```

Comment:
Would be nice to have a comment or README-style file explaining how this was generated

Reply:
I agree that we should include a source, but I don't think there's too much we need to say about how we read them and saved them in `npz` format. How about this?
```suggestion
    # Test matrices from `illc1850.coord` and `mhd1280b.cua` distributed with
    # PROPACK 2.1: http://sun.stanford.edu/~rmunk/PROPACK/
    relative_path = "propack_test_data.npz"
```

Reply:
Yep, LGTM